### DISCUSSION TITLE
Link prediction issues

### DISCUSSION THREAD

**Discussion by OmonkeyGOD** [pyg-team/pytorch_geometric#9921] [Q&A]
Hello, I am trying to use Geometric for link prediction and have some questions I hope you can help me with. 

1. I manually generate train/val/test datasets without using RandomLinkSplit. Does it make sense if I only do negative sampling for supervision links in the training set? For example, I have 200 links, and 160/20/20 for train/val/test purpose. 100 of the training links would be used for message passing. Then, simply sample around 60 negative links, combining the remaining 60 links for supervision. 
2. During the validation step, does it make sense to use all positive training edges in the model for message passing, and then use all links (20 positive and ~20 negatively sampled) in the val set for prediction? And for testing step, using train + val pos links for message passing and only 20 pos test link to predict the accuracy? Are negative links necessary in testing step?
